servant-server/test/Servant/ServerSpec.hs

```diff
@@ -667,3 +668,41 @@ errorsSpec = do
       nf <> he `shouldBe` he
       nf <> ib `shouldBe` ib
       nf <> wm `shouldBe` wm
+
+type MiscCombinatorsAPI
+  =    "version" :> HttpVersion :> Get '[JSON] String
+  :<|> "secure"  :> IsSecure :> Get '[JSON] String
+  :<|> "host"    :> RemoteHost :> Get '[JSON] String
+  :<|> "vault"   :> Vault :> Get '[JSON] ()
+
+miscApi :: Proxy MiscCombinatorsAPI
+miscApi = Proxy
+
+miscServ :: Server MiscCombinatorsAPI
+miscServ = versionHandler
+      :<|> secureHandler
+      :<|> hostHandler
+      :<|> vaultHandler
+
+  where versionHandler = return . show
+        secureHandler Secure = return "secure"
+        secureHandler NotSecure = return "not secure"
+        hostHandler = return . show
+        vaultHandler _ = return ()
```

Comment:
I'd prefer either not having the vault test at all, or having it test *something* ;) If you thinking just seq-ing the vault doesn't actually make a difference, then we maybe shouldn't have the test?